### PR TITLE
fix incorrect compiler activation

### PR DESCRIPTION
**Pull request by h-vetinari**
Follow-up from #102, see [here](https://github.com/conda-forge/ctng-compiler-activation-feedstock/pull/102#discussion_r1520749310)

### CODE REVIEW

**Review by xhochy — APPROVED** [review]
All the `*_2` builds produced packages without a `run_exports`. We should ensure that these builds will have one before we merge again.

**Review by xhochy — CHANGES_REQUESTED** [review]
All the *_2 builds produced packages without a run_exports. We should ensure that these builds will have one before we merge again.

**Comment by beckermr** [comment]
I guess we should download the artifacts to be sure?

Maybe a conda build bug?

**Comment by beckermr** [comment]
I think maybe we need to nest run exports under build for outputs. I wonder if doing things this old way is broken.

**Comment by xhochy** [comment]
> I think maybe we need to nest run exports under build for outputs. I wonder if doing things this old way is broken.

Going to investigate that locally, just waiting on a very long rerender to finish.

**Comment by isuruf** [comment]
> I think maybe we need to nest run exports under build for outputs. I wonder if doing things this old way is broken.

Yeah, that's probably it. The last PR introduced new `build` key for the outputs and the top level `run_exports` does not work and need a `build: run_exports` key instead. To confirm, just try `conda render recipe` on the following,

```yaml
package:
  name: foo
  version: 3

build:
  number: 0

outputs:
  - name: bar
    build:
      skip: true   # [osx]
    run_exports:
      - bar >=3
    requirements:
      host:
        - zlib

  - name: baz
    run_exports:
      - baz >=3
    requirements:
      host:
        - zlib
```
and you'll see run_exports are missing on bar, but it does exist on baz.

**Comment by xhochy** [comment]
Locally, the `run_exports` look fine now.

**Comment by beckermr** [comment]
Should we add a smithy lint for any run exports not under build?

**Comment by xhochy** [comment]
> Should we add a smithy lint for any run exports not under build?

Yes, that would be helpful.

**Review by xhochy — APPROVED** [review]
Verified locally for a package that we have `run_exports` again-

**Comment by beckermr** [comment]
Shall we merge?

File: conda-forge.yml [hunk 1]
```diff
@@ -1,8 +1,9 @@
+azure:
+  store_build_artifacts: true
```
Comment (h-vetinari) [inline conda-forge.yml:2]: Was this left in intentionally?

**Review by h-vetinari — COMMENTED** [review]
Thanks everyone for helping fix this!

File: conda-forge.yml (hunk 1 above)
Comment (xhochy) [reply conda-forge.yml:2]: No :(

Comment (h-vetinari) [reply conda-forge.yml:2]: Fixed in https://github.com/conda-forge/ctng-compiler-activation-feedstock/commit/9d65b2e5d051628c2b604fe54105ba76cc57927d so we don't forget next time around.

**Pull request merged**